Compilation error when using the USE_ROCM=on flag.

### Describe the bug

```
clang-15: error: no such file or directory: '/public/home/aisi/users/denghui/abacus/soft/OpenBLAS/lib/libopenblas.so;-lpthread;-lm;-ldl'
make[2]: *** [CMakeFiles/abacus_pw.dir/build.make:721: abacus_pw] Error 1
make[2]: Leaving directory '/public/home/aisi/users/denghui/abacus/abacus-develop-2024-04-28/abacus-develop/build'
make[1]: *** [CMakeFiles/Makefile2:919: CMakeFiles/abacus_pw.dir/all] Error 2
make[1]: Leaving directory '/public/home/aisi/users/denghui/abacus/abacus-develop-2024-04-28/abacus-develop/build'
make: *** [Makefile:153: all] Error 2
```

### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

```
Currently Loaded Modulefiles:
  1) compiler/devtoolset/7.3.1   2) compiler/rocm/dtk-23.10     3) mpi/hpcx/2.11.0/gcc-7.3.1   4) compiler/cmake/3.17.2
```

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [X] Verify the issue is not a duplicate.
- [X] Describe the bug.
- [X] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -330,6 +330,7 @@ endif()
 # version. Use cmake with AMD-ROCm:
 # https://rocmdocs.amd.com/en/latest/Installation_Guide/Using-CMake-with-AMD-ROCm.html
 if(USE_ROCM)
+  cmake_minimum_required(VERSION 3.18)
   if(NOT DEFINED ROCM_PATH)
     set(ROCM_PATH
         "/opt/rocm"

